pkg/tournament/tournament.go

```diff
@@ -852,126 +917,77 @@ func StartAllRoundCountdowns(ctx context.Context, ts TournamentStore, id string,
 	t.Lock()
 	defer t.Unlock()
 
-	if t.IsFinished {
-		return fmt.Errorf("tournament %s has finished", id)
+	err = startTournamentChecks(t)
+	if err != nil {
+		return err
 	}
 
-	if len(t.Divisions) == 0 {
-		return fmt.Errorf("cannot start tournament %s with no divisions", t.Name)
+	for division := range t.Divisions {
+		err = startDivisionChecks(t, division, round)
+		if err != nil {
+			return err
+		}
 	}
 
 	for division := range t.Divisions {
-		dm := t.Divisions[division].DivisionManager
-		if dm == nil {
-			return fmt.Errorf("cannot start round %d for division %s because it has a nil division manager", round, division)
-		}
-		if dm.GetDivisionControls().GameRequest == nil {
-			return fmt.Errorf("no division game controls have been set for division %v", division)
-		}
-		isReady, err := dm.IsRoundReady(round)
+		err := t.Divisions[division].DivisionManager.StartRound(false)
 		if err != nil {
 			return err
 		}
-		if !isReady {
-			return fmt.Errorf("cannot start round %d for division %s because the round is not ready", round, division)
-		}
+	}
+	t.IsStarted = true
+	err = ts.Set(ctx, t)
+	if err != nil {
+		return err
 	}
 
 	for division := range t.Divisions {
-		t.IsStarted = true
-		err := StartRoundCountdown(ctx, ts, id, division, round, false, true, false)
+		err := sendDivisionStart(ts, t.UUID, division, round)
 		if err != nil {
-			t.IsStarted = false
 			return err
 		}
 	}
-	return ts.Set(ctx, t)
-}
 
-// DivisionChannelName returns a channel name that can be used
-// for sending communications regarding a tournament and division.
-func DivisionChannelName(tid, division string) string {
-	// We encode to b64 because division can contain spaces.
-	return base64.URLEncoding.EncodeToString([]byte(tid + ":" + division))
+	return nil
 }
 
 func StartRoundCountdown(ctx context.Context, ts TournamentStore, id string,
-	division string, round int, save, send, lock bool) error {
+	division string, round int) error {
 	t, err := ts.Get(ctx, id)
 	if err != nil {
 		return err
 	}
 
-	if lock {
-		t.Lock()
-		defer t.Unlock()
-	}
-
-	if t.IsFinished {
-		return fmt.Errorf("tournament %s has finished", id)
-	}
-
-	divisionObject, ok := t.Divisions[division]
+	t.Lock()
+	defer t.Unlock()
 
-	if !ok {
-		return fmt.Errorf("division %s does not exist", division)
+	err = startTournamentChecks(t)
+	if err != nil {
+		return err
 	}
 
-	if divisionObject.DivisionManager == nil {
-		return fmt.Errorf("division %s does not have enough players or controls to set pairings", division)
+	err = startDivisionChecks(t, division, round)
+	if err != nil {
+		return err
 	}
 
-	/*	We will deal with auto start at a later date
-		if manual && divisionObject.Controls.AutoStart && divisionObject.DivisionManager.IsStarted() {
-			return fmt.Errorf("division %s has autostart enabled and cannot be manually started", division)
-		}*/
-
-	if !t.IsStarted {
```

Comment:
it seems the new `StartRoundCountdown` no longer requires `t.IsStarted`. is that intended?

the rest of the PR seems ok.

Reply:
Yes, that was intentional. I realized that if we keep that check and some of the rounds on SARC fail, then we have no way to individually start the rounds that had failed, since t.Started was never set to true. In this way we can regard SARC is a batch call identical to calling SRC individually.

Reply:
it almost is, but SARC also sets t.IsStarted. if you call each SRC individually then t.IsStarted will never be set and that'd break other stuffs (and you can no longer call SARC because they'd be at different rounds). we don't have a "just set t.IsStarted" as a separate step. we probably should, and then both SARC and SRC can check that this has been done. not sure.

Reply:
Ah, good point. I think the most straightforward way is to just have both SRC and SARC set t.Started = true on success.